pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -244,3 +432,109 @@ func (r *MachineSyncReconciler) shouldMirrorCAPIMachineToMAPIMachine(ctx context
+func (r *MachineSyncReconciler) fetchCAPIInfraResources(ctx context.Context, capiMachine *capiv1beta1.Machine) (client.Object, client.Object, error) {
+	var infraCluster, infraMachine client.Object
+
+	infraClusterKey := client.ObjectKey{
+		Namespace: capiMachine.Namespace,
+		Name:      capiMachine.Spec.ClusterName,
+	}
+
+	infraMachineRef := capiMachine.Spec.InfrastructureRef
+	infraMachineKey := client.ObjectKey{
+		Namespace: infraMachineRef.Namespace,
+		Name:      infraMachineRef.Name,
+	}
+
+	switch r.Platform {
+	case configv1.AWSPlatformType:
+		infraCluster = &capav1beta2.AWSCluster{}
+		infraMachine = &capav1beta2.AWSMachine{}
+	case configv1.PowerVSPlatformType:
+		infraCluster = &capibmv1.IBMPowerVSCluster{}
+		infraMachine = &capibmv1.IBMPowerVSMachine{}
+	default:
+		return nil, nil, fmt.Errorf("%w: %s", errPlatformNotSupported, r.Platform)
+	}
+
+	if err := r.Get(ctx, infraClusterKey, infraCluster); err != nil {
+		return nil, nil, fmt.Errorf("failed to get CAPI infrastructure cluster: %w", err)
+	}
+
+	if err := r.Get(ctx, infraMachineKey, infraMachine); err != nil {
+		return nil, nil, fmt.Errorf("failed to get CAPI infrastructure machine: %w", err)
+	}
+
+	return infraCluster, infraMachine, nil
+}
+
+// capiInfraMachineIsEqual checks whether the provided CAPI infra machines are equal.
+func capiInfraMachineIsEqual(platform configv1.PlatformType, infraMachine1, infraMachine2 client.Object) (bool, error) {
+	switch platform {
+	case configv1.AWSPlatformType:
+		typedInfraMachine1, ok := infraMachine1.(*capav1beta2.AWSMachine)
+		if !ok {
+			return false, errAssertingCAPIAWSMachine
+		}
+
+		typedinfraMachine2, ok := infraMachine2.(*capav1beta2.AWSMachine)
+		if !ok {
+			return false, errAssertingCAPIAWSMachine
+		}
+
+		return reflect.DeepEqual(typedInfraMachine1.Spec, typedinfraMachine2.Spec) && util.ObjectMetaIsEqual(typedInfraMachine1.ObjectMeta, typedinfraMachine2.ObjectMeta), nil
+	case configv1.PowerVSPlatformType:
+		typedInfraMachine1, ok := infraMachine1.(*capibmv1.IBMPowerVSMachine)
+		if !ok {
+			return false, errAssertingCAPIIBMPowerVSMachine
+		}
+
+		typedinfraMachine2, ok := infraMachine2.(*capibmv1.IBMPowerVSMachine)
+		if !ok {
+			return false, errAssertingCAPIIBMPowerVSMachine
+		}
+
+		return reflect.DeepEqual(typedInfraMachine1.Spec, typedinfraMachine2.Spec) && util.ObjectMetaIsEqual(typedInfraMachine1.ObjectMeta, typedinfraMachine2.ObjectMeta), nil
+	default:
+		return false, fmt.Errorf("%w: %s", errPlatformNotSupported, platform)
+	}
+}
+
+// updateSynchronizedConditionWithPatch updates the synchronized condition
+// using a server side apply patch. We do this to force ownership of the
+// 'Synchronized' condition and 'SynchronizedGeneration'.
+func (r *MachineSyncReconciler) updateSynchronizedConditionWithPatch(ctx context.Context, mapiMachine *machinev1beta1.Machine, status corev1.ConditionStatus, reason, message string, generation *int64) error {
```

Comment:
Why not `applySynchronizedCondition`?

Does this controller make any other writes to the object? We may want the synchronized condition to use its own field owner otherwise other writes could remove the condition, especially when we start copying status back and forth

Reply:
> applySynchronizedCondition

Happy to change to this though, I think it makes more sense.

Reply:
> Does this controller make any other writes to the object? We may want the synchronized condition to use its own field owner otherwise other writes could remove the condition, especially when we start copying status back and forth

It does resync the object, so yeah, probably nothing else on the status side at the moment, but there will be

Reply:
Ah, IIRC this was named like this because initially I used a patch, rather than SSA. Must've not updated helper func names :)

Reply:
Lets change this, and I think use a specific name for the field manager for the synchronized condition so that in the future, if we start writing the status otherwise, the synchronized condition can still be treated as a separate transaction for the purposes of SSA

Reply:
Renamed func and added a more specific field owner.